montecarlo/cfast_mcarlo.py

```diff
@@ -514,6 +522,30 @@ def _locate_in_the_middle(self, fire_room):
 
         return location
 
+    def _locate_randomly(self, fire_room):
+        location = {}
+        # locate fire in compartment
+        compa = self.s.query("SELECT * FROM aamks_geom WHERE name=?", (fire_room, ))[0]
+
+        # draw location (uniform across the compartment)
+        dx = uniform(0, compa['width'])
+        dy = uniform(0, compa['depth'])
+
+        x = int(compa['x0']+dx)
+        y = int(compa['y0']+dy)
+        self._fire_height = round((compa['height'] * 0.9 * (1-math.log10(uniform(1,9)))) / 100, 2)
+        z = int(self._fire_height * 100) + compa['height'] * int(compa['floor'])
```

Comment:
here, if there is a situation that the rooms on the lower floors have a higher or lower height than the room on the current floor, these calculations will be wrong

Reply:
OK, to be corrected.